### PR TITLE
Kubeconfig context chooser, UI tests

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Refactor kk upload, add context chooser
  - To create kubeconfig with multiple contexts:
```
export KUBECONFIG=<path_to_kk_1>:<path_to_kk_2>:...:<path_to_kk_n>, n ∈ ℕ
kc config view --flatten > kubeconfig--combined.yaml
```
- UI tests for
   - Invalid kubeconfig - both for text paste and upload
   - Enkode URL - one with kubeconfig (with and without token), one without (manual kubeconfig upload in Busola)
   - Other login tests - test uploading kubeconfig without token and manually adding a token

Considerations
- I've mocked out enkode website - we can use real one when we integrate it in Busola.
- No OIDC tests for now

**Related issue(s)**

### CODE REVIEW

File: tests/tests/invalid-kubeconfig.spec.js [hunk 1]
```diff
@@ -2,20 +2,43 @@
...
 
+    cy.getIframeBody()
+      .contains('Paste')
+      .click();
+
     cy.getIframeBody()
       .find('#textarea-kubeconfig')
-      .type('wrong_kubeconfig');
+      .type('wrong_kubeconfig')
+      .getIframeBody()
```
Comment (Sawthis) [inline tests/tests/invalid-kubeconfig.spec.js:18]: It should be separate chain cy.getIframeBoddy()...

File: tests/tests/invalid-kubeconfig.spec.js [hunk 2]
```diff
@@ -2,20 +2,43 @@
...
-      .contains('Apply kubeconfig')
+      .find('[role=alert][aria-label="invalid-kubeconfig"]')
+      .shouldHaveTrimmedText('Error reading kubeconfig');
+  });
+
+  it('Use wrong kubeconfig - from file', () => {
+    cy.visit(config.clusterAddress);
+    cy.get('[data-testid=app-switcher]')
+      .click()
+      .get('[data-testid=addcluster]')
```
Comment (Sawthis) [inline tests/tests/invalid-kubeconfig.spec.js:31]: It should be separate chain

File: tests/tests/login-enkode.spec.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      cy.visit(`${config.clusterAddress}?init=${params}`);
+
+      cy.url().should('match', /namespaces$/);
+      cy.getIframeBody()
+        .find('thead')
+        .should('be.visible');
+    });
+  });
+
+  it('Only params from enkode, upload kubeconfig manually', () => {
```
Comment (Sawthis) [inline tests/tests/login-enkode.spec.js:22]:
```suggestion
  it('Only params from enkode without kubeconfig', () => {
```

File: tests/tests/login-enkode.spec.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        .contains('Testing Busola')
+        .click();
+
+      cy.getLeftNav()
+        .contains('Test nav node')
+        .should('be.visible');
+    });
+  });
+
+  it('Login with no token in kubeconfig', () => {
```
Comment (Sawthis) [inline tests/tests/login-enkode.spec.js:52]:
```suggestion
  it('Login without token in kubeconfig', () => {
```

**Review by dariadomagala — CHANGES_REQUESTED** [review]
1. Check this flow:
* Add a Kubeconfig file
* Switch tab to `paste`
* Go pack to `upload` tab
* My uploaded file is there yet, the button is disabled. Adding the file again doesn't help
The same thing happens if I start with the `paste` tab

2. We don't have enough data here to show the context dropdown and inputs like issuer etc. 
Screen Shot 2021-06-02 at 08 10 16

3. 
Screen Shot 2021-06-02 at 08 14 43
Blank page with the error in the console appeared after I uploaded an invalid kubeconfig

4. If I apply combined kubeconfig using init params, I get only the first context, no option to choose which one I'd want to use

**Review by Sawthis — CHANGES_REQUESTED** [review]

**Review by Sawthis — APPROVED** [review]

**Review by Sawthis — DISMISSED** [review]

**Review by Sawthis — APPROVED** [review]

**Review by dariadomagala — COMMENTED** [review]
After choosing a context it jums back to the `apply kubeconfig` view for 1-2sec, could we avoid it?

**Review by dariadomagala — APPROVED** [review]

**Pull request merged**